### PR TITLE
refactor(checkbox): tailwindify

### PR DESCRIPTION
**Pull request by julio8a**
Supports #1500

### CODE REVIEW

File: src/components/calcite-checkbox/calcite-checkbox.scss [hunk 1]
```diff
@@ -58,9 +58,7 @@
   }
```
Comment (jrowlingson) [inline src/components/calcite-checkbox/calcite-checkbox.scss:58]: Possibly a question for @asangma - it looks like this mixin is only used once, should we kill? Create a util for this?

**Comment by jrowlingson** [comment]
The checkbox size expressed in terms of `fontSize` seems a little weird.. there is also two different patterns in here for setting properties based on the `scale` attribute. I have'nt played around with this but wonder if we could revisit later on.. perhaps remove the reliance on grid?

**Comment by eriklharper** [comment]
> perhaps remove the reliance on grid?

Since we don't have to worry about IE anymore, I think we can still take advantage of css grid here, but we could definitely refactor how the size of the checkbox is expressed.  These two issues are mutually exclusive I'd say.

**Comment by julio8a** [comment]
Okay, can I merge this and focus those mentioned updates later?

**Review by jrowlingson — APPROVED** [review]

**Pull request merged**